src/agent/metrics.rs

```diff
@@ -0,0 +1,376 @@
+
+        let shared_state = Arc::new(Mutex::new(server));
+
+        let dashboard_route = warp::path("dashboard")
+            .and(warp::path::end())
+            .and_then(move || {
+                let shared_state = shared_state.clone();
+                async move {
+                    let locked_state = shared_state.lock().await;
+                    let response = locked_state
+                        .render_dashboard()
+                        .await
+                        .unwrap_or_else(|e| {
+                            // Add logging here
+			    error!(locked_state.logger,"Could not render dashboard"; "error" => e.to_string());
+
+                            // Withhold failure details from client
+                            "Could not render dashboard!".to_owned()
+                        });
+                    Result::<Box<dyn Reply>, Rejection>::Ok(Box::new(reply::with_status(
+                        reply::html(response),
+                        StatusCode::OK,
+                    )))
+                }
+            });
+
+        warp::serve(dashboard_route).bind(addr).await;
+    }
+
+    /// Create an HTML view of store data
+    async fn render_dashboard(&self) -> Result<String, Box<dyn std::error::Error>> {
+        // Prepare response channel for request
+        let (local_tx, local_rx) = oneshot::channel();
+        let (global_data_tx, global_data_rx) = oneshot::channel();
+        let (global_metadata_tx, global_metadata_rx) = oneshot::channel();
+
+        // Request price data from local store
+        self.local_store_tx
+            .send(Message::LookupAllPriceInfo {
+                result_tx: local_tx,
+            })
+            .await?;
+
+        self.global_store_lookup_tx
+            .send(Lookup::LookupAllAccountsData {
+                result_tx: global_data_tx,
+            })
+            .await?;
+
+        self.global_store_lookup_tx
+            .send(Lookup::LookupAllAccountsMetadata {
+                result_tx: global_metadata_tx,
+            })
+            .await?;
+
+        // Await the results
+        let local_data = local_rx.await?;
+        let global_data = global_data_rx.await??;
+        let global_metadata = global_metadata_rx.await??;
+
+        let symbol_view =
+            build_dashboard_data(local_data, global_data, global_metadata, &self.logger);
+
+        let mut uptime_seconds = self.start_time.elapsed().as_secs();
+
+        // Shave bigger units off the seconds value. Decide the count
+        // and replace with remainder.
+        let uptime_days = uptime_seconds / (24 * 3600); // 24h
+        uptime_seconds %= 24 * 3600;
+
+        let uptime_hours = uptime_seconds / 3600; // 1h
+        uptime_seconds %= 3600;
+
+        let uptime_minutes = uptime_seconds / 60; // 1min
+        uptime_seconds %= 60;
+
+        let uptime_string = format!(
+            "{}d{}h{}m{}s",
+            uptime_days, uptime_hours, uptime_minutes, uptime_seconds
+        );
+
```

Comment:
You can use use `humantime::format_duration` to just get the same thing:

```rust
use humantime::format_duration;
let duration_string = format_duration(duration);
```

Reply:
Done 🫡